pkg/controller/main-controller.go

```diff
@@ -734,7 +734,10 @@ func (c *Controller) syncHandler(key string) error {
 	}
 
 	// Check if we are decommissioning a pool before we ensure defaults, as that would populate a defaulted pool name
-	tenant, err = c.checkForPoolDecommission(ctx, key, tenant, tenantConfiguration)
+	temporal_tenant, err := c.checkForPoolDecommission(ctx, key, tenant, tenantConfiguration)
```

Comment:
No _ underscore naming allowed in Go.

Reply:
This still won't solve the problem for the future. The function here incorrectly  returns nil, nil

I believe the correct fix is #1625